rts/motoko-rts/src/visitor.rs

```diff
@@ -3,13 +3,16 @@ use crate::types::*;
 
 /// A visitor that passes field addresses of fields with pointers to dynamic heap to the given
 /// callback
-pub unsafe fn visit_pointer_fields<F>(
+pub unsafe fn visit_pointer_fields<C, F, G>(
+    ctx: &mut C,
     obj: *mut Obj,
     tag: Tag,
     heap_base: usize,
-    mut visit_ptr_field: F,
+    visit_ptr_field: F,
+    visit_field_range: G,
```

Comment:
Could you please describe the contract of the new callback in the comments?

Reply:
See d10949a956c08f3bf11d5e0eb383ac80c48ce278.